_posts/2023-06-11-ayb-helping-you-own-your-data.md

```diff
@@ -0,0 +1,140 @@
+------------+-------
+ PostgreSQL | 10 
+ SQLite     | 9 
+ DuckDB     | 9 
+
+Rows: 3
+```
+
+The command line invocations above are a thin wrapper around `ayb`'s HTTP API. Here are the same commands as above, but with `curl`:
+```bash
+$ curl -w "\n" -X POST http://127.0.0.1:5433/v1/marcua -H "entity-type: user"
+
+{"entity":"marcua","entity_type":"user"}
+
+$ curl -w "\n" -X POST http://127.0.0.1:5433/v1/marcua/test.sqlite -H "db-type: sqlite"
+
+{"entity":"marcua","database":"test.sqlite","database_type":"sqlite"}
+
+$ curl -w "\n" -X POST http://127.0.0.1:5433/v1/marcua/test.sqlite/query -d 'CREATE TABLE favorite_databases(name varchar, score integer);'
+
+{"fields":[],"rows":[]}
+
+$ curl -w "\n" -X POST http://127.0.0.1:5433/v1/marcua/test.sqlite/query -d "INSERT INTO favorite_databases (name, score) VALUES (\"PostgreSQL\", 10);"
+
+{"fields":[],"rows":[]}
+
+$ curl -w "\n" -X POST http://127.0.0.1:5433/v1/marcua/test.sqlite/query -d "INSERT INTO favorite_databases (name, score) VALUES (\"SQLite\", 9);"
+
+{"fields":[],"rows":[]}
+
+$ curl -w "\n" -X POST http://127.0.0.1:5433/v1/marcua/test.sqlite/query -d "INSERT INTO favorite_databases (name, score) VALUES (\"DuckDB\", 9);"
+
+{"fields":[],"rows":[]}
+
+$ curl -w "\n" -X POST http://127.0.0.1:5433/v1/marcua/test.sqlite/query -d "SELECT * FROM favorite_databases;"
+
+{"fields":["name","score"],"rows":[["PostgreSQL","10"],["SQLite","9"],["DuckDB","9"]]}
+```
+
+## Students, sharers, and sovereigns 
+If `ayb` is successful, it will become easier to create a database, interact with it, and share it with relevant people/organizations. There are three groups that would benefit most from such a tool, and by studying the problems they face, we can make `ayb` more useful for them.
+
+**Students**. The barrier to learn how to work with data is too high, and much of it is in operational overhead (how to set up a database? how to connect to it? how to share what I've learned?). Ideally, aside from registering for an account and creating a database, there should be no operational overhead to writing your first SQL query. It should be easy to fork a data set and start asking questions, and it should be easy to start inserting rows into your own small data set. If you get stuck, it should be easy to give a mentor or teacher access to your database and get some help.
+
+**Sharers**. Scientists, journalists, and other people who want to share a data set have largely ad hoc means to share that data, and their collaborators and readers' experience is limited by the ad hoc sharing decisions. You've encountered this if you've ever tried to do something with the CSV file someone shared over email or if you've wanted to visualize the data presented in an article in a slightly different way. Sharers should be able to create a database, add collaborators, and eventually open it up to the public to fork/query in a read-only way with as little overhead for themselves or the recipient as possible.
```

Comment:
It might be nice for comparison to call out the differences with just sharing a data file vs ayb's data interface. The data file is nice in that consumers pay for their own compute, but ayb lets consumers automatically get updated data, etc.

Reply:
Good call! Added a note about the tradeoffs! :)